### PR TITLE
Add code related to registration, key policies, instance policies and allowed IP port

### PR DESCRIPTION
**Pull request by thu2thu2**
Added SDK support related to registration, key policies, instance policies and allowed IP port.

Also added more examples to README.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -151,7 +193,50 @@ then configure key protect service with authenticator. then run operations on se
...
+```
+
+### Disable a key
+```
+            InputStream inputstream = new FileInputStream(<PATH_TO_Disable_Key_Body>)
+            ActionOnKeyOptions disableKeyOptionsModel = new ActionOnKeyOptions.Builder()
+                    .id(keyId)
+                    .bluemixInstance(bluemixInstance)
+                    .action("disable")
+                    .keyActionOneOf(inputstream)
```
Comment (mrodden) [inline README.md:218]: I don't think there should be a key body for disable...?

File: README.md [hunk 2]
```diff
@@ -151,7 +193,50 @@ then configure key protect service with authenticator. then run operations on se
...
+```
+
+### Enable a key
+```
+            InputStream inputstream = new FileInputStream(<PATH_TO_Enable_Key_Body>)
+            ActionOnKeyOptions enableKeyOptionsModel = new ActionOnKeyOptions.Builder()
+                    .id(keyId)
+                    .bluemixInstance(bluemixInstance)
+                    .action("enable")
+                    .keyActionOneOf(inputstream)
```
Comment (mrodden) [inline README.md:232]: Same with enable

File: README.md [hunk 3]
```diff
@@ -170,6 +255,380 @@ then configure key protect service with authenticator. then run operations on se
...
+
+###### Set automatic rotation policy for a key
+```
+            // Construct an instance of the CollectionMetadata model
+            CollectionMetadata collectionMetadataModel = new CollectionMetadata.Builder()
+                    .collectionType("application/vnd.ibm.kms.policy+json")
+                    .collectionTotal(Long.valueOf("1"))
+                    .build();
+
+            // Construct an instance of the KeyPolicyRotationRotation model
```
Comment (mrodden) [inline README.md:284]: Add an additional comment or text here to say that we are building a policy that will cause keys to be automatically rotated every 2 months.

File: README.md [hunk 4]
```diff
@@ -170,6 +255,380 @@ then configure key protect service with authenticator. then run operations on se
...
+                    .bluemixInstance(bluemixInstance)
+                    .setKeyPoliciesOneOf(setKeyPoliciesOneOfModel)
+                    .policy("dualAuthDelete")
+                    .build();
+
+            // Invoke operation with valid options model (positive test)
+            Response<GetKeyPoliciesOneOf> response = testService.putPolicy(putPolicyOptionsModel).execute();
+            GetKeyPoliciesOneOf responseObj = response.getResult();
+```
+###### Set multiple policies for a key
```
Comment (mrodden) [inline README.md:350]:
```suggestion
###### Set multiple policies for a key

Here we set up a Dual Auth delete policy and also a policy to rotate the key every 2 months, using one HTTP request.
```

File: README.md [hunk 5]
```diff
@@ -170,6 +255,380 @@ then configure key protect service with authenticator. then run operations on se
...
+###### Set allowed network policy for an instance
+
+```
+            // Construct an instance of the CollectionMetadata model
+            CollectionMetadata collectionMetadataModel = new CollectionMetadata.Builder()
+                    .collectionType("application/vnd.ibm.kms.policy+json")
+                    .collectionTotal(Long.valueOf("1"))
+                    .build();
+
+            // Construct an instance of the InstancePolicyAllowedNetworkPolicyDataAttributes model
```
Comment (mrodden) [inline README.md:471]:
```suggestion
            // Construct an instance of the InstancePolicyAllowedNetworkPolicyDataAttributes model
            // Here we are building a network policy that allows access to the instance via public and private networks (the default)
```

**Review by mrodden — CHANGES_REQUESTED** [review]

File: README.md [hunk 6]
```diff
@@ -151,7 +193,50 @@ then configure key protect service with authenticator. then run operations on se
...
-### Creating/Retrieving an Import token
+
+### Rotate a key
+```
+            InputStream inputstream = new FileInputStream(<PATH_TO_Rotate_Key_Body>)
+            ActionOnKeyOptions actionOnKeyOptionsModel = new ActionOnKeyOptions.Builder()
+                    .id(keyId)
+                    .bluemixInstance(bluemixInstance)
+                    .action("rotate")
+                    .keyActionOneOf(inputstream)
```
[the remark this answers is not in the document]
Comment (thu2thu2) [reply README.md:204]: This is for rotating a key so the key id is required. See https://cloud.ibm.com/apidocs/key-protect?code=java#rotatekey

File: README.md (hunk 1 above)
Comment (thu2thu2) [reply README.md:218]: Agreed. Fixed code to make keyActionOneOf nullable.

**Comment by thu2thu2** [comment]
Created new branch and opened new PR after restructuring the project into 1 module.

**Pull request closed without merging**